### PR TITLE
Changes for rsc app installation support in v1.0

### PR DESCRIPTION
**Pull request by sweta-thapliyal**

### CODE REVIEW

**Review by AkJo — APPROVED** [review]

**Comment by Danielabom** [comment]
Contacted author to get info.

File: api-reference/v1.0/resources/teamsapppermissionset.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: "teamsAppPermissionSet resource type"
+description: "Set of Resource specific permissions."
```
Comment (Danielabom) [inline api-reference/v1.0/resources/teamsapppermissionset.md:3]:
```suggestion
description: "Set of resource specific permissions."
```

File: changelog/Microsoft.Teams.GraphSvc.json [hunk 2]
```diff
@@ -58,6 +58,64 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [teamsAppResourceSpecificPermission](https://learn.microsoft.com/en-us/graph/api/resources/teamsAppResourceSpecificPermission?view=graph-rest-1.0) resource type.",
+          "Target": "teamsAppResourceSpecificPermission"
+        },
+        {
+          "Id": "3685f48e-7a82-4fa4-b313-ce909435e6a1",
+          "ApiChange": "Property",
+          "ChangedApiName": "authorization",
+          "ChangeType": "Addition",
+          "Description": "Added the `authorization` property to the [teamsAppDefinition](https://learn.microsoft.com/en-us/graph/api/resources/teamsAppDefinition?view=graph-rest-1.0) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.Teams.GraphSvc.json:100]:
```suggestion
          "Description": "Added the **authorization** property to the [teamsAppDefinition](https://learn.microsoft.com/en-us/graph/api/resources/teamsAppDefinition?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.Teams.GraphSvc.json [hunk 3]
```diff
@@ -58,6 +58,64 @@
...
+          "Target": "teamsAppDefinition"
+        },
+        {
+          "Id": "3685f48e-7a82-4fa4-b313-ce909435e6a1",
+          "ApiChange": "Property",
+          "ChangedApiName": "consentedPermissionSet",
+          "ChangeType": "Addition",
+          "Description": "Added the `consentedPermissionSet` property to the [teamsAppInstallation](https://learn.microsoft.com/en-us/graph/api/resources/teamsAppInstallation?view=graph-rest-1.0) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.Teams.GraphSvc.json:108]:
```suggestion
          "Description": "Added the **consentedPermissionSet** property to the [teamsAppInstallation](https://learn.microsoft.com/en-us/graph/api/resources/teamsAppInstallation?view=graph-rest-1.0) resource.",
```

File: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
+title: "teamsAppInstallation in team: upgrade"
 description: "Upgrade an app installation in a team"
 author: "akjo"
 ms.localizationpriority: medium
 ms.prod: "microsoft-teams"
 doc_type: apiPageType
 ---
 
-# teamsAppInstallation: upgrade
+# teamsAppInstallation in team: upgrade
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-teamsappinstallation-upgrade.md:10]:
```suggestion
# teamsAppInstallation in a team: upgrade
```

File: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md [hunk 5]
```diff
@@ -1,31 +1,34 @@
 ---
-title: "teamsAppInstallation: upgrade"
+title: "teamsAppInstallation in team: upgrade"
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-teamsappinstallation-upgrade.md:2]:
```suggestion
title: "teamsAppInstallation in a team: upgrade"
```

File: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md [hunk 6]
```diff
@@ -1,31 +1,34 @@
...
-# teamsAppInstallation: upgrade
+# teamsAppInstallation in team: upgrade
 
 Namespace: microsoft.graph
 
 Upgrade an [app installation](../resources/teamsappinstallation.md) in a [team](../resources/team.md) to the latest version of the app.
 
-> **Note:** Currently, this operation does not support upgrade of apps that require resource-specific consent permissions. For details, see [Known issues](/graph/known-issues#Installation-of-apps-that-require-resource-specific-consent-permissions-is-not-supported).
+> **Note:**
+> - Currently, upgrading an app that requires resource-specific consent permissions is not supported in application context.
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-teamsappinstallation-upgrade.md:17]:
```suggestion
> - Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.
```

File: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md [hunk 7]
```diff
@@ -1,31 +1,34 @@
...
+> - Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.
 
 ## Permissions
 
 One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
 
 |Permission type      | Permissions (from least to most privileged)              |
 |:--------------------|:---------------------------------------------------------|
-|Delegated (work or school account) | TeamsAppInstallation.ReadWriteSelfForTeam, TeamsAppInstallation.ReadWriteForTeam, Group.ReadWrite.All**, Directory.ReadWrite.All** |
+|Delegated (work or school account) | TeamsAppInstallation.ReadWriteSelfForTeam<sup>1</sup>, TeamsAppInstallation.ReadWriteForTeam<sup>1</sup>, TeamsAppInstallation.ReadWriteAndConsentForTeam, TeamsAppInstallation.ReadWriteAndConsentSelfForTeam, Group.ReadWrite.All<sup>2</sup>, Directory.ReadWrite.All<sup>2</sup> |
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-teamsappinstallation-upgrade.md:25]: Are these new permissions? If so, please add changelog entries.

File: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md [hunk 8]
```diff
@@ -1,31 +1,34 @@
...
-# teamsAppInstallation: upgrade
+# teamsAppInstallation in a team: upgrade
 
 Namespace: microsoft.graph
 
 Upgrade an [app installation](../resources/teamsappinstallation.md) in a [team](../resources/team.md) to the latest version of the app.
 
-> **Note:** Currently, this operation does not support upgrade of apps that require resource-specific consent permissions. For details, see [Known issues](/graph/known-issues#Installation-of-apps-that-require-resource-specific-consent-permissions-is-not-supported).
+> **Note:**
+> - Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-teamsappinstallation-upgrade.md:17]:
```suggestion
> Currently, upgrading an app that requires resource-specific consent permissions is not supported in an application context.
```

File: api-reference/v1.0/api/team-post-installedapps.md [hunk 9]
```diff
@@ -13,19 +13,22 @@ Namespace: microsoft.graph
 
 Install an [app](../resources/teamsapp.md) to the specified [team](../resources/team.md).
 
-> **Note:** Currently, this operation does not support installation of apps that require resource-specific consent permissions. For details, see [Known issues](/graph/known-issues#Installation-of-apps-that-require-resource-specific-consent-permissions-is-not-supported).
+> **Note:**
+> - Currently, installing an app that requires resource-specific consent permissions is not supported in application context.
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-post-installedapps.md:17]:
```suggestion
> Currently, installing an app that requires resource-specific consent permissions is not supported in an application context.
```

File: api-reference/v1.0/api/team-post-installedapps.md [hunk 10]
```diff
@@ -42,25 +45,33 @@ POST /teams/{team-id}/installedApps
 
 ## Request body
 
+The request body should contain the catalog app's generated app ID. For details, see [teamsApp properties](../resources/teamsapp.md#properties).
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-post-installedapps.md:48]:
```suggestion
The request body should include the generated app ID of the catalog app. For more information, see [teamsApp properties](../resources/teamsapp.md#properties).
```

File: api-reference/v1.0/api/team-post-installedapps.md [hunk 11]
```diff
@@ -42,25 +45,33 @@ POST /teams/{team-id}/installedApps
...
+The following table lists additional properties that can be included in the request body.
+
 | Property   | Type |Description|
 |:---------------|:--------|:----------|
-|teamsApp|String|The id of the app to add.|
+|teamsApp|String|The ID of the app to add.|
+|consentedPermissionSet|[teamsAppPermissionSet](../resources/teamsapppermissionset.md)|The set of resource-specific permissions that are being consented to.|
+
+> **Note**:
+> The permissions consented to during the install must be the same as the resource-specific permissions present in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-post-installedapps.md:58]:
```suggestion
> The permissions consented to during the installation must match the resource-specific permissions defined in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are specified in the **teamsAppDefinition**, permissions can be omitted in the request body.
```

File: api-reference/v1.0/api/chat-teamsappinstallation-upgrade.md [hunk 12]
```diff
@@ -1,31 +1,33 @@
...
+title: "teamsAppInstallation in chat: upgrade"
 description: "Update an app installed in a chat and bring it in sync with the current version available in the tenant app catalog."
 author: "subray"
 ms.localizationpriority: high
 ms.prod: "microsoft-teams"
 doc_type: apiPageType
 ---
 
-# teamsAppInstallation: upgrade
+# teamsAppInstallation in chat: upgrade
```
Comment (Danielabom) [inline api-reference/v1.0/api/chat-teamsappinstallation-upgrade.md:10]:
```suggestion
# teamsAppInstallation in a chat: upgrade
```

File: api-reference/v1.0/api/team-teamsappinstallation-upgrade.md [hunk 13]
```diff
@@ -41,15 +44,24 @@ POST /teams/{team-id}/installedApps/{app-installation-id}/upgrade
...
+In the request body, supply a JSON representation of the parameters.
+
+The following table shows additional parameters that can be used with the upgrade action.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|consentedPermissionSet|[teamsAppPermissionSet](../resources/teamsapppermissionset.md)|Set of resource specific permissions that are being consented.|
+
+> **Note**:
+> The permissions consented to during the upgrade must be the same as the resource-specific permissions present in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.
```
Comment (Danielabom) [inline api-reference/v1.0/api/team-teamsappinstallation-upgrade.md:56]:
```suggestion
> The permissions consented to during the installation must match the resource-specific permissions defined in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.
```

File: api-reference/v1.0/api/chat-teamsappinstallation-upgrade.md [hunk 14]
```diff
@@ -34,21 +36,35 @@ One of the following permissions is required to call this API. To learn more, in
...
+In the request body, supply a JSON representation of the parameters.
+
+The following table lists additional parameters that can be used with the upgrade action.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|consentedPermissionSet|[teamsAppPermissionSet](../resources/teamsapppermissionset.md)|The set of resource-specific permissions that are being consented to.|
+
+> **Note**:
+> The permissions consented to during the upgrade must be the same as the resource-specific permissions present in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.
```
Comment (Danielabom) [inline api-reference/v1.0/api/chat-teamsappinstallation-upgrade.md:50]:
```suggestion
> he permissions consented to during the installation must match the resource-specific permissions defined in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). IIf only delegated resource-specific permissions are specified in the **teamsAppDefinition**, permissions can be omitted in the body of this request.
```

File: api-reference/v1.0/api/chat-post-installedapps.md [hunk 15]
```diff
@@ -47,22 +48,29 @@ POST /chats/{chat-id}/installedApps
 
 ## Request body
 
-The request body should contain the catalog app's generated app ID. For details, see [teamsApp properties](../resources/teamsapp.md#properties).
+The request body should contain the catalog generated app ID for the app catalog. For details, see [teamsApp properties](../resources/teamsapp.md#properties).
```
Comment (Danielabom) [inline api-reference/v1.0/api/chat-post-installedapps.md:51]:
```suggestion
The request body should contain the catalog generated app ID for the app catalog. For details, see [teamsApp properties](../resources/teamsapp.md#properties).
```
```suggestion
The request body should contain the catalog generated app ID for the app catalog. For details, see [teamsApp properties](../resources/teamsapp.md#properties).
```
```suggestion
The request body should include the generated app ID of the catalog app. For more information, see [teamsApp properties](../resources/teamsapp.md#properties).
```

File: api-reference/v1.0/api/chat-post-installedapps.md [hunk 16]
```diff
@@ -47,22 +48,29 @@ POST /chats/{chat-id}/installedApps
...
+The following table lists additional parameters that can be used with the request body.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|consentedPermissionSet|[teamsAppPermissionSet](../resources/teamsapppermissionset.md)|Set of resource-specific permissions that are being consented to.|
+
+> **Note**: The permissions consented to during the install must be same as the resource-specific permissions present in the [teamsAppDefinition](../resources/teamsAppDefinition.md) of the app. To get the application and delegated resource-specific permissions, see [Example 7](../api/appcatalogs-list-teamsapps.md#example-7-list-applications-with-a-given-id-and-return-only-the-resource-specific-permissions-required-by-the-app). If only delegated resource-specific permissions are present in **teamsAppDefinition**, permissions can be omitted in the body of this request.
```
Comment (Danielabom) [inline api-reference/v1.0/api/chat-post-installedapps.md:58]:
```suggestion
> **Note**: 
```

**Review by Danielabom — CHANGES_REQUESTED** [review]
@sweta-thapliyal, thank you for working on this PR. Please see my comments. Quick question are you adding new permissions, or is it more of a doc update because the operations did not reflect the correct permissions? If these are new permissions, we need changelog entries.

File: api-reference/v1.0/api/chat-teamsappinstallation-upgrade.md (hunk 12 above)
Comment (sweta-thapliyal) [reply api-reference/v1.0/api/chat-teamsappinstallation-upgrade.md:10]: All the other teams related titles have similar structure, where we don't use article. Let's keep this one consistent.

**Review by Danielabom — APPROVED** [review]
PR approved.

File: api-reference/v1.0/resources/teamsappauthorization.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+ms.localizationpriority: medium
+ms.prod: "microsoft-teams"
+doc_type: resourcePageType
+---
+
+# teamsAppAuthorization resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
```
Comment (sweta-thapliyal) [inline api-reference/v1.0/resources/teamsappauthorization.md:14]:
```suggestion
```

File: api-reference/v1.0/resources/teamsapppermissionset.md [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+ms.localizationpriority: medium
+ms.prod: "microsoft-teams"
+doc_type: resourcePageType
+---
+
+# teamsAppPermissionSet resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
```
Comment (sweta-thapliyal) [inline api-reference/v1.0/resources/teamsapppermissionset.md:14]:
```suggestion
```

File: api-reference/v1.0/resources/teamsappresourcespecificpermission.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+ms.localizationpriority: medium
+ms.prod: "microsoft-teams"
+doc_type: resourcePageType
+---
+
+# teamsAppResourceSpecificPermission resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
```
Comment (sweta-thapliyal) [inline api-reference/v1.0/resources/teamsappresourcespecificpermission.md:14]:
```suggestion
```

**Pull request merged**